crates/core/c8y_api/src/utils.rs

```diff
@@ -13,6 +14,38 @@ pub mod bridge {
             Err(_err) => false,
         }
     }
+
+    pub fn is_c8y_bridge_established(message: &Message) -> bool {
+        match message.payload_str() {
+            Ok(payload) => {
+                message.topic.name == C8Y_BRIDGE_HEALTH_TOPIC
+                    && (payload == C8Y_BRIDGE_UP_PAYLOAD || payload == C8Y_BRIDGE_DOWN_PAYLOAD)
+            }
+            Err(_err) => false,
+        }
+    }
+}
+
+pub mod tedge_agent {
```

Comment:
This `c8y_api` crate really feels like the wrong place for this logic. This logic should exist somewhere in the mapper as only the `c8y_mapper` cares about the `tedge-agent`. The `c8y_api` doesn't.

Reply:
Agreed, I moved inside the c8y_mapper_actor itself.